lib/puppet/application.rb

```diff
@@ -281,6 +281,23 @@ def run_mode( mode_name = nil)
       @run_mode = Puppet::Util::RunMode[ mode_name || Puppet.settings.preferred_run_mode ]
     end
 
+    # Sets environment_mode name
+    # @param mode_name [Symbol] The name of the environment mode to run in. May
+    #   be one of :local, :remote, or :not_required. This impacts where the
+    #   application looks for its specified environment. If :not_required or
+    #   :remote are set, the application will not fail if the environment does
+    #   not exist on the local filesystem.
+    def environment_mode(mode_name)
```

Comment:
A rubism for naming setters and getters is to call this method `environment_mode=` and the one below `environment_mode`

Reply:
oh, perfect, thank you!

Reply:
We want this to be a DSL mode, so this gets to stay as is